### PR TITLE
Firmware generation tool

### PR DESCRIPTION
**Pull request by VLitanADI**
implements #411

### CODE REVIEW

File: tools/calibration-96tof1/cal_eeprom/cal_eeprom.py [hunk 1]
```diff
@@ -34,471 +34,53 @@
...
-import sys
-from natsort import natsorted, ns
-import re
-import os
-import pandas as pd
-from . import firmware_gen as lf
-import json 
-import aditofpython as tof
-import tof_calib.device as device
+# import os
```
Comment (dNechita) [inline tools/calibration-96tof1/cal_eeprom/cal_eeprom.py:40]: Is this still required?

File: tools/calibration-96tof1/cal_eeprom/cal_map.py [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+
+                sub_packet_value = struct.unpack('<f', r_b[j:j+4])
+                j = j+4
+                sub_packet_value = int(sub_packet_value[0])
+                i = i + 1
+                parameter_size = sub_packet_value
+                #print("Param Size", parameter_size)
+
+                number_of_elements = int(parameter_size/4)  # 4:size of float
+                #print("Number of elements", number_of_elements)
```
Comment (dNechita) [inline tools/calibration-96tof1/cal_eeprom/cal_map.py:363]: Are all these prints required to be here?

**Review by dNechita — APPROVED** [review]
Looks good.

One thing that would be nice is to add a short description at the beginning of each script that explains the purpose of the script (what it can do and why it is useful).

File: tools/calibration-96tof1/cal_eeprom/cal_eeprom.py (hunk 1 above)
Comment (VLitanADI) [reply tools/calibration-96tof1/cal_eeprom/cal_eeprom.py]: fixed.

File: tools/calibration-96tof1/cal_eeprom/cal_map.py (hunk 2 above)
Comment (VLitanADI) [reply tools/calibration-96tof1/cal_eeprom/cal_map.py]: not sure. I turned them into debug logs as they might be useful

**Pull request merged**